Tests/RequestsSpec/requestParserSpec.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+import Quick
+import Nimble
+import Requests
+
+class RequestParseSpec: QuickSpec {
+    override func spec() {
+        describe("#RequestParser") {
+            var httpParser: RequestParser!
+
+            beforeEach {
+                httpParser = RequestParser()
+            }
+
+            it ("throws an empty request error if the request is empty") {
+                let emptyRequest = String()
+                expect{try httpParser.parse(request: emptyRequest)}.to(throwError(RequestParserError.EmptyRequest))
+            }
+
+            it ("throws an invalid status line error if the request format is incorrect") {
+                let badRequest = """
+                Host: localhost:5000
+                User-Agent: curl/7.54.0
+                Accept: */*
+                """
+                expect {try httpParser.parse(request: badRequest)}.to(throwError(RequestParserError.InvalidStatusLine(badRequest)))
+            }
+
+            it ("returns an HttpRequest object if the incoming request is valid") {
+                let validRequest = """
+                    GET /logs HTTP/1.1
+                    Host: localhost:5000
+                    Connection: Keep-Alive
+                    User-Agent: Apache-HttpClient/4.3.5 (java 1.5)
+                    Accept-Encoding: gzip,deflate
+                """
+
+                let expected = HttpRequest(
+                        method: "GET",
+                        url: "/logs",
+                        version: "HTTP/1.1",
+                        headers: [
+                            "Host": "localhost:5000",
+                            "Connection": "Keep-Alive",
+                            "User-Agent": "Apache-HttpClient/4.3.5 (java 1.5)",
+                            "Accept-Encoding": "gzip,deflate"
+                        ]
+                )
+
+                do {
+                    let parsedRequest = try httpParser.parse(request: validRequest)
+                    expect(parsedRequest).to(equal(expected))
+                } catch {
```

Comment:
I'm not very familiar with swift, but is this common to have an empty catch block?

Reply:
I am not sure. At first, I put something in there to make the test failed if it ever enters the catch block. Then I saw someone leaves it empty. It stills works, so I just go with empty. I don't  fully understand the consequence of my choice yet. @cmvandrevala any thought?